src/plugins/data/public/search/legacy/call_msearch.ts

```diff
@@ -39,9 +39,10 @@ import { LegacyFetchHandlers } from '../../../common/search/search_source';
  * @internal
  */
 export function getCallMsearch({ http }: { http: HttpStart }): LegacyFetchHandlers['callMsearch'] {
-  return async ({ body, signal }) => {
+  return async ({ body, signal, dataSourceId }) => {
```

Comment:
would this require modifying the  msearch search params? if we have some?

Reply:
It's doable to move the data source id to request body. Then we don't need to add extra data source id query.